lib/fast_gettext/lazy_mo_file.rb

```diff
@@ -0,0 +1,28 @@
+require 'fast_gettext/vendor/mofile'
+module FastGettext
+  # Responsibility:
+  #  - abstract mo files with lazy loading for Mo Repository
+  class LazyMoFile < MoFile
+    def initialize(file)
+      @filename = file
+    end
+
+    def load_data(file)
+      if file.is_a? FastGettext::GetText::MOFile
+        @data = file
+      else
+        @data = FastGettext::GetText::MOFile.open(file, "UTF-8")
+      end
+      make_singular_and_plural_available
+    end
+
+    def [](key)
+      data[key]
+    end
+
+    def data
+      load_data(@filename) if @data.nil?
```

Comment:
I made `load_data` private and kept it in place. `||=` construct would be possible but I'm afraid it wouldn't add much elegance since we need to call `make_singular_and_plural_available` when the data gets loaded.